Src/IronPython/Compiler/Ast/SequenceExpression.cs

```diff
@@ -18,13 +18,15 @@
 
 namespace IronPython.Compiler.Ast {
     public abstract class SequenceExpression : Expression {
-        private readonly Expression[] _items;
+        protected readonly Expression[] _items;
 
         protected SequenceExpression(Expression[] items) {
             _items = items;
         }
 
-        public IList<Expression> Items => _items;
+        public IReadOnlyList<Expression> Items => _items;
+
+        internal Expression[] UnsafeItems => _items; // TODO: get rid of this
```

Comment:
Good luck with this! 😄  `IReadOnlyCollection` should have had `CopyTo`, just like `ICollection` has. I suppose `ArrayUtils.Insert<T>(T item, IReadOnlyList<T> list)` can also do `if (list is T[] array) ...` to maintain performance .

Reply:
Unfortunately `ArrayUtils.Insert` is in the DLR and I didn't want to rev it just for this. I guess as long as we don't start using `UnsafeItems` all over the place (I don't see why we would) this is fine.